app/controllers/api/v2/domains_controller.rb

```diff
@@ -88,6 +95,10 @@ def destroy
       def allowed_nested_id
         %w(subnet_id location_id organization_id)
       end
+
+      def verify_proxy_id(id)
+        id.nil? || SmartProxy.authorized(:view_smart_proxies).find_by_id(id).present?
```

Comment:
But we want only `nil` here, we don't want to verify dns_id if it was not passed as attribute (`domain_parameters[:dns_id]` is `nil`), if user sent empty string we want to trigger the verification and try load the proxy so it fails and tells user that empty string is invalid value.